### DISCUSSION TITLE
road to v0.3

### DISCUSSION THREAD

**Discussion by amitu** [fastn-stack/fastn#501] [General]
This is the official tracker for what is to be done before v0.3 is ready.

- [Cheatsheet](https://github.com/ftd-lang/ftd/blob/main/Cheatsheet.md)
- [ChangeLog](https://github.com/ftd-lang/ftd/blob/main/Changelog.md)

# doc-site blockers

- [ ] https://github.com/ftd-lang/ftd/discussions/295
  - [x] basic implementation
  - [x] CHILDREN-COUNT: len(a), $a.length 
  - [ ] Color-doc build issue - needed for all color schemes
  - [ ] fonts: import fonts - e.g. `fpm.font-copy: $inter.fonts.Roboto`
id=580%3A222485&t=m7Zq7zJwXT5zPk4z-0)
- [ ] font weights in design system
- [ ] there is extra gap in headings, remove it
- [ ] [re-export](https://github.com/ftd-lang/ftd/discussions/443) ([why it is a blocker](https://github.com/ftd-lang/ftd/discussions/462))
- [ ] https://github.com/ftd-lang/ftd/discussions/548
- [ ] [ftd.ui as kernal component](https://github.com/ftd-lang/ftd/discussions/559)
- [ ] local storage
- [ ] inbuilt clamp function
- [ ] [missing css attributes](https://github.com/ftd-lang/ftd/issues/492)

<details>
<summary>Done</summary>

- [x] [SIBLING-INDEX: $LOOP.COUNTER](https://github.com/ftd-lang/ftd/discussions/558)
- [x] design system update: update variable names as per [figma](https://www.figma.com/file/8JNdHNa41JDHbjjUhKU5aW/Fastn-Master-Component-File?node-
- [x] cleanup: merge spacing and spacing-mode
- [x] sticky
- [x] border-style
- [x] id
- [x] ftd.checkbox
- [x] conditional image src
- [x] `ftd.decimal`, `ftd.input`
- [x] `on-input`,`on-change`, `on-focus`, `on-blur`
- [ ] `ftd.code`, `ftd.iframe`
- [x] processors
- [x] Default functions (`ftd.toggle`, `ftd.increment`, `ftd.set-value`)
- [x] [`ftd.breakpoints-width`](https://github.com/ftd-lang/ftd/discussions/506)
- [x] `ftd.device` to become an `or-type` 
- [x] `import fpm`
- [x] `-` not working in FScript
- [x] Component common property: `region: h1, h2.. h6`
- [x] [css reset](https://github.com/ftd-lang/ftd/issues/215)
- [x] ftd.device variable
- [x] [role is pending](https://github.com/ftd-lang/ftd/discussions/506)
- [x] Events
- [x] `fpm build --edition 2022`
- [x] Create fastn-template
- [x] [ftd.responsive-length](https://github.com/ftd-lang/ftd/discussions/508)
- [x] `inherited.color` and `inherited.type` global variables
- [x] children can not be empty right now
- [x] [or-type constant construction shorthand (only short-hand allowed)](https://github.com/ftd-lang/ftd/discussions/470#discussioncomment-4502389)

</details>

# kameri app blocker

- [ ] `http` function should accept the JSON/assignment type data in FScript
- [ ] how to set enum?

# optimisation

- dependency graph optimisation
- reduce inline styles

# kerala team blocker

- [ ] dynamic dom
- [ ] #503
- [ ] #504 
- [ ] https://github.com/ftd-lang/fpm/discussions/689
- [ ] inline f-script

# regression

- markup

# good to have

- [ ] https://github.com/ftd-lang/ftd/discussions/295
- [ ] any type
- [ ] hoisting variable declarations (var dec not allowed inside component definition, but everywhere else, vars will be hoisted to module scope)
- [x] [resize](https://developer.mozilla.org/en-US/docs/Web/CSS/resize) (so our layout related docs are better)

# FPM Processors

[All Example](https://github.com/ftd-lang/fpm/tree/main/examples)

- http
- toc
- get-data
- sitemap
- full-sitemap
- request-data
- document-readers
- document-writers
- user-groups
- user-group-by-id
- get-identities
- document-id
- document-full-id
- document-suffix
- document-name
- user-details
- fpm-apps
- is-reader

**Comment by gsalunke** [comment]
@amitu Please, add this also:

- Improve colors of fastn default color scheme

**Comment by amitu** [comment]
The purpose of this discussion is over.
